.github/workflows/on-update.yml

```diff
@@ -8,6 +8,10 @@ on:
   pull_request:
     types: [opened, synchronize, reopened]
 
+concurrency:
+  group: ${{ github.event_name == 'pull_request' && format('on-update-pr-{0}', github.event.pull_request.number) || format('on-update-run-{0}', github.run_id) }}
+  cancel-in-progress: ${{ github.event_name == 'pull_request' }}
```

Comment:
Checking whether it's a pull request is redundant, when non-pull-requests are in unique groups.